### PR TITLE
Actually send Angular errors to Sentry

### PR DESCRIPTION
**Pull request by theefer**
Was having a quick look and it turns out because Angular errors don't bubble to the window, they were never sent to Sentry. Doh.

This code snippet is borrowed from Composer, and tested locally. In theory we could try to use [angular-raven](https://github.com/gdi2290/angular-raven), but in practice it doesn't give us much more than the 7 lines of code below (and more configuration hassle).

### CODE REVIEW

File: kahuna/public/js/sentry/sentry.js [hunk 1]
```diff
@@ -3,20 +3,18 @@ import raven from 'raven-js';
...
-sentry.factory('sentry', [function() {
-
-    function trigger(errorMessage, extra) {
-        raven.captureException(new Error(errorMessage), { extra });
-    }
-
-    return { trigger };
+// TODO: Alternatively could investigate angular-raven
+sentry.config(['$provide', function ($provide) {
+    $provide.decorator('$exceptionHandler', ['$delegate', function ($delegate) {
```
Comment (jamesgorrie) [inline kahuna/public/js/sentry/sentry.js]: Does this capture http errors too? If so, it probably shouldn't.

**Comment by jamesgorrie** [comment]
> we could try to use angular-raven

I haven't really found any that give that much value either.
Nice find :+1:.

File: kahuna/public/js/sentry/sentry.js (hunk 1 above)
Comment (theefer) [inline kahuna/public/js/sentry/sentry.js]: Good catch, it is.

I was hoping to be able to do something clever where only uncaught HTTP error would bubble, unfortunately **all** throw exceptions (as opposed to returning `$q.reject(...)`) bubble to the global error handler no matter whether you catch them or not, and that includes HTTP errors.

I've added a fix in another commit.

**Comment by theefer** [comment]
I seem to be getting a fair number of 429 when testing locally, not entirely sure why.

**Comment by jamesgorrie** [comment]
Ah - to sentry? Is that a pay for thing?

**Comment by jamesgorrie** [comment]
:+1: to the new commit

**Comment by theefer** [comment]
We pay for Sentry.

**Pull request merged**